src/main/java/de/theit/jenkins/crowd/CrowdConfigurationService.java

```diff
@@ -204,8 +209,23 @@ public CrowdConfigurationService(String url, String applicationName, Secret pass
         this.nestedGroups = pNestedGroups;
         this.useSSO = useSSO;
         this.useCache = useCache;
+        this.useTokenCache = useTokenCache;
         this.cacheSize = cacheSize;
         this.cacheTTL = cacheTTL;
+
+
+        if(cacheSize != null) {
+
+            this.isGroupMemberCache = new CacheMap<>(cacheSize);
+            this.userFromSSOTokenCache = new CacheMap<>(cacheSize);
+            this.userCache = new CacheMap<>(cacheSize);
+            this.groupCache = new CacheMap<>(cacheSize);
+            this.authoritiesForUserCache = new CacheMap<>(cacheSize);
+            if(useTokenCache) {
+                this.validationCache = new CacheMap<>(cacheSize);
+            }
+    }
```

Comment:
1. Please fix formatting
3. If `useTokecCache` will be unchecked, and then Jenkins will reboot, we have no possibility to enable it again without a restart. Line 942 will crash in this situation.
4. It's better to remove if case for `this.validationCache`, allow it to be always initiated as other caches
6. Nice catch 👍

Reply:
@akouznetchik please apply only points 1, 2, 4 thanks.